### DISCUSSION TITLE
How to use static images in multiple apps?

### DISCUSSION THREAD

**Discussion by krckyboy** [vercel/turborepo#4977] [Help] [answered]
Ideally, I want to have a root `/public/images` directory where I would store images that are used in multiple `/apps`, however that doesn't seem to work as my apps expect `/public` folder to be within `apps` themselves (`/apps/public/images` and not `/public/images`). 

How should I approach this if I want to use same static images across apps in both `<img>` and `scss` files?

**Comment by JasperAlexander** [comment]
Have you tried creating a package for that?

**Reply by krckyboy** [reply]
> Have you tried creating a package for that?

It's been a while, but I believe I tried to do that, but it didn't seem to work. I might be in the wrong, but I don't think it worked due to the fact that we'd be pointing to assets, not actual code. Anyway, I investigated what would be easy options for this to work, and came up with symlinking `/public/images` to each individual app in `/apps`:

```bash
root_dir="$(pwd)"  # Set the project's root directory to the current working directory
apps_dir="$root_dir/apps"

# Check if the apps directory exists
if [ ! -d "$apps_dir" ]; then
  echo "Apps directory $apps_dir does not exist. Please make sure your apps are located in the correct directory."
  exit 1
fi

# Iterate over the apps directory and create symbolic links
for app_dir in "$apps_dir"/*
do
  app=$(basename "$app_dir")
  target_dir="$root_dir/public/images"
  symlink_dir="$app_dir/public"

  # Create the necessary directories if they don't exist
  mkdir -p "$symlink_dir"

  if [ -L "$symlink_dir/images" ]; then
    echo "Symbolic link $symlink_dir/images already exists. Skipping $app"
    continue
  fi

  ln -s "$target_dir" "$symlink_dir/images"
  echo "Created symbolic link: $symlink_dir/images -> $target_dir"

  # Sync the changes in the root directory to the symlinked directory
  rsync -av --delete "$target_dir/" "$symlink_dir/images"
  echo "Synced changes in root public/images to $symlink_dir/images"
done

```

**Comment by ekafyi** [comment]
There is also a good example of shell script run as postinstall command in this turborepo starter: https://github.com/oneezy/monorepo

**Comment by anthonyshew** [comment, marked as answer]
Going to close this one as a duplicate of https://github.com/vercel/turbo/discussions/5046

Let's chat there!
